common/rpc/encryption/localStoreCertProvider.go

```diff
@@ -25,30 +25,38 @@
 package encryption
 
 import (
+	"crypto/md5"
 	"crypto/tls"
 	"crypto/x509"
 	"encoding/base64"
+	"encoding/pem"
 	"errors"
 	"fmt"
 	"io/ioutil"
 	"sync"
+	"time"
 
 	"go.temporal.io/server/common/service/config"
 )
 
 var _ CertProvider = (*localStoreCertProvider)(nil)
 var _ ClientCertProvider = (*localStoreCertProvider)(nil)
+var _ CertExpirationChecker = (*localStoreCertProvider)(nil)
 
 type localStoreCertProvider struct {
 	sync.RWMutex
 
 	tlsSettings       *config.GroupTLS
 	workerTLSSettings *config.WorkerTLS
 
-	serverCert *tls.Certificate
-	clientCert *tls.Certificate
-	clientCAs  *x509.CertPool
-	serverCAs  *x509.CertPool
+	serverCert          *tls.Certificate
+	clientCert          *tls.Certificate
+	clientCAs           *x509.CertPool
+	serverCAs           *x509.CertPool
+	serverCAsWorker     *x509.CertPool
+	clientCACerts       []*x509.Certificate // copies of certs in the clientCAs CertPool for expiration checks
+	serverCACerts       []*x509.Certificate // copies of certs in the serverCAs CertPool for expiration checks
+	serverCACertsWorker []*x509.Certificate // copies of certs in the serverCAs CertPool for expiration checks
```

Comment:
[Nit] last comment should read "copies of certs in the serverCAsWorker CertPool for expiration checks"

Reply:
also, does the pool not support iterating through its certificates? (maybe ill find out later in the PR why this extra slice copy was needed)